src/Crawler.php

```diff
@@ -284,50 +237,20 @@ public function startCrawling($baseUrl)
 
     protected function startCrawlingQueue()
     {
+        $fulfilledHandler = $this->getFulfilledHandler();
```

Comment:
Rename this `onFulfill` and use it immediately in the pool. Same for the `failedHandler`, rename that to `onFail`